### PR TITLE
Table + Containers - Handle hiding, and fixup containers code.

### PR DESCRIPTION
**Pull request by mfalesni**
Handle hiding of the table. Phase #1 involves the .rows() and the things that use it directly. The other things will eventually need this change too once needed.

{{pytest: cfme/tests/containers/test_relationships_tables.py -v --long-running}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit cd24ff74a333d42fe58f74823b5a78d6d7afec61:

`cfme/web_ui/__init__.py`:
- :red_circle: Line 62:1: **F401** *'CNoSuchElementException' imported but unused*
- :red_circle: Line 409:51: **F821** *undefined name 'NoSuchElementException'*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 24aa23dc2c29bbf6baa0fb9f989ed3b088131fcb:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c6199a1bd3174ed7b81d04f30d30ccd76a6a246c:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by RonnyPfannschmidt** [comment]
looks good

the @landscape-bot is drunk

**Comment by mfalesni** [comment]
@RonnyPfannschmidt I added a couple of other changes, yelling when it cannot find the row when clicking a single row and also ./ in the xpath expressions

**Comment by dajoRH** [comment]
Lint report for commit 815d73032b83f30318c4132142b1e33e3a4922b1:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fa6d00416fa761f9c549dd532e1044aa2b26b0f6:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5f937c3f6f4ecf92e31c8e05fc5d1303fa23e583:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 1c842ae326fd95cb2240163429cf54f54272b62f:

`cfme/containers/project.py`:
- :bangbang: Line 15:17: **E126** *continuation line over-indented for hanging indent*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9a3f57008027f6f6685b5ec9c55d4eeed99a56a9:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by RonnyPfannschmidt** [comment]
with the reduction of contexts to single element mappings, would it make sense to move to using the single value as context? @psav @jkrocil

**Comment by mfalesni** [comment]
@RonnyPfannschmidt I don't think that would work for all use cases, however a simplification could be done, I was thinking about it:

You could just pass objects as `*args` and the objects' classes would define into what key it should be converted, it would generate the context dictionary automatically then.

```python
class Foo(object):
    context_name = 'foo'

foo = Foo()
force_navigate('bar', foo)

# inside navigation
args = [<Foo instance 0xbar>]
# will be converted to
args = {'foo': <Foo instance 0xbar>}
```

**Comment by dajoRH** [comment]
Lint report for commit 4805b4569ca2ece39d5c5a2c9a6cdbcc789d2a77:

`cfme/web_ui/__init__.py`:
- :red_circle: Line 71:1: **F811** *redefinition of unused 'attributize_string' from line 68*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 79d91b1d95eb3d15f76762935abcb4e5a104e268:

`cfme/web_ui/__init__.py`:
- :red_circle: Line 71:1: **F811** *redefinition of unused 'attributize_string' from line 68*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a29f0acbebc5c6e47f071ccb457f336fedf290cc:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d5a570bbf32eb4f0996c8a982bd7f2f5e7b3d6bc:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 739e2fd9ced71114670478a7de74bc4277bd1d11:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fd819d5e8bbfdadcf8ab19c05f0a38bb6ff0cc85:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/containers/container.py [hunk 1]
```diff
@@ -1,35 +1,27 @@
...
 nav.add_branch(
     'containers_containers',
     {
         'containers_container':
-        [
-            lambda ctx: list_tbl.select_row_by_cells(
-                {'Name': ctx['container'].name, 'Pod Name': ctx['pod'].name}),
-            {
-                'containers_container_edit_tags':
-                lambda _: pol_btn('Edit Tags'),
```
Comment (psav) [inline cfme/containers/container.py]: We are removing navigation points here......is that a good idea?

File: cfme/containers/container.py [hunk 2]
```diff
@@ -52,8 +44,7 @@ def navigate(self, detail=True):
                     'containers_container_detail', context={
                         'container': self, 'pod': self.pod})
         else:
-            sel.force_navigate('containers_container',
-                               context={'container': self, 'pod': self.pod})
```
Comment (psav) [inline cfme/containers/container.py]: pod was added erroneously?

File: cfme/containers/image.py [hunk 3]
```diff
@@ -67,8 +57,8 @@ def navigate(self, detail=True):
             if not self._on_detail_page():
                 sel.force_navigate(
                     'containers_image_detail', context={
-                        'image': self, 'provider': self.provider})
```
Comment (psav) [inline cfme/containers/image.py]: were all these extra context items added erroneously

File: cfme/containers/container.py (hunk 1 above)
Comment (mfalesni) [inline cfme/containers/container.py]: They are not used in the code

File: cfme/containers/image.py (hunk 3 above)
Comment (mfalesni) [inline cfme/containers/image.py]: No. The thing is that the nav steps were badly designed. They asked for 2 things despite one thing providing the other :) So I removed that and made the nav steps use the dependency.

**Comment by jkrocil** [comment]
Looks good; 5.6 has a single failure that is not caused by this PR.

**Pull request merged**